zerver/lib/soft_deactivation.py

```diff
@@ -201,6 +204,11 @@ def add_missing_messages(user_profile: UserProfile) -> None:
                 # We are going to short circuit this iteration as its no use
                 # iterating since user unsubscribed before soft-deactivation
                 continue
+
+        if stream.is_support_stream() and not check_access_based_on_can_access_stream_topics_group(
+            user_profile, stream
+        ):
+            continue
```

Comment:
I don't know if this is the right approach but I couldn't think of anything else. One thing that came to my mind was doing something like this:
```
subscription_logs = list(
        RealmAuditLog.objects.filter(
            modified_user=user_profile, 
            modified_stream_id__in=stream_ids, 
            modified_stream__can_access_stream_topics_group_id=everyone_group_id,
            event_type__in=events
        )
        .order_by("event_last_message_id", "id")
        .only("id", "event_type", "modified_stream_id", "event_last_message_id")
    )
```

But this is wrong because this won't create `UserMessage` rows for a soft deactivated user who is in `can_access_stream_topics_group` in a support stream.

Reply:
This is also increasing the query count.

Reply:
I just noticed that this doesn't account for the scenario where a user who is not a member of `can_access_stream_topics_group` receives messages in a topic they created while they are soft-deactivated. Ideally, we should create `UserMessage` rows for all messages sent in that topic. However, this would require fetching the first message in the topic to identify the sender, which could result in additional queries. Should we proceed with this approach despite the extra queries?

Reply:
Right I will ask CZO.

Reply:
Based on the [CZO](https://chat.zulip.org/#narrow/stream/6-frontend/topic/Support.20Streams/near/1918573) discussion the soft-reactivation should skip support streams.